infra/cron/src/jobs/consistency.js

```diff
@@ -0,0 +1,100 @@
+const checker = require('@origin/consistency')
+const fs = require('fs')
+const tmp = require('tmp')
+const sendgridMail = require('@sendgrid/mail')
+const esmImport = require('esm')(module)
+const contractsContext = esmImport('@origin/graphql/src/contracts').default
+const { setNetwork } = esmImport('@origin/graphql/src/contracts')
+const logger = require('../logger')
+
+function todaysDateString() {
+  const now = new Date()
+  const year = now.getUTCFullYear()
+  const month = String(now.getUTCMonth()).padStart(2, '0')
+  const day = String(now.getUTCDate()).padStart(2, '0')
+  return `${year}-${month}-${day}`
+}
+
+/**
+ * A job to run consistency checker and send an E-mail with results
+ * @param job
+ * @returns {Promise<*>}
+ */
+async function ConsistencyCheckJob(job) {
+  logger.info(`Processing ConsistencyCheckJob id=${job.id} data= ${job.data}`)
+  job.progress(0)
+
+  if (typeof process.env.PROVIDER_URL === 'undefined') {
+    throw new Error('Env var PROVIDER_URL must be set')
+  }
+  if (typeof process.env.EMAILS_TO === 'undefined') {
+    throw new Error('Warning: EMAILS_TO env var is not set')
+  }
+  if (typeof process.env.SENDGRID_API_KEY === 'undefined') {
+    throw new Error('Env var SENDGRID_API_KEY must be set')
+  }
+  if (typeof process.env.SENDGRID_FROM_EMAIL === 'undefined') {
+    logger.warn('Warning: SENDGRID_FROM_EMAIL env var is not set')
+  }
+
+  const logFile = tmp.fileSync()
+  const network = process.env.NETWORK || 'mainnet'
+
+  setNetwork(network)
+  if (!contractsContext.web3) {
+    throw new Error('web3 not initialized')
+  }
+  console.log(`Logging to file ${logFile.name}`)
+  const checkCount = await checker.main({
+    network,
+    web3: contractsContext.web3,
+    jsonRPCURL: process.env.PROVIDER_URL,
+    ipfsGateway: process.env.IPFS_GATEWAY || 'https://ipfs.originprotocol.com',
+    fromBlock: 6425000,
+    identity: true,
+    listings: true,
+    offers: true,
+    logFile: logFile.name
+  })
+
+  logger.info(`${checkCount} checks run`)
+
+  job.progress(90)
+
+  const checkerOutput = await fs.readFileSync(logFile.fd)
+  const checkerOutputEncoded = Buffer.from(checkerOutput).toString('base64')
+
+  job.progress(92)
+
+  sendgridMail.setApiKey(process.env.SENDGRID_API_KEY)
+
+  // Assemble the E-mail
+  const to = process.env.EMAILS_TO.split(',').map(s => s.trim())
+  const dateString = todaysDateString()
+  const email = {
+    to,
+    from: process.env.SENDGRID_FROM_EMAIL,
+    subject: `Daily data consistency checker results (${dateString}) (${network})`,
+    text: `Attached are the results of the ${network} consistency checker run on ${dateString}`,
```

Comment:
Might be good to put in the text some super high level stats. because I bet at some point we won't look at the attachment. We could also change the subject in case some discrepancy found rather than none.
All these are nice-to-have improvements but absolutely not blockers for rolling this out.

Reply:
Agreed, but think it should be done after this PR.  Added #2861.